projects/step-core/src/lib/components/editable-textarea-label/editable-textarea-label.component.ts

```diff
@@ -0,0 +1,56 @@
+import { ChangeDetectorRef, Component, ElementRef, forwardRef, ViewChild } from '@angular/core';
+import { NG_VALUE_ACCESSOR } from '@angular/forms';
+import { EditableComponent } from '../../shared/editable-component';
+
+const DEFAULT_TEXTAREA_ROWS = 4;
+const LINE_HEIGHT = 18;
+const PADDING_TOP_BOTTOM = 8;
+const BORDER_WIDTH = 1;
+
+@Component({
+  selector: 'step-editable-textarea-label',
+  templateUrl: './editable-textarea-label.component.html',
+  styleUrls: ['./editable-textarea-label.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => EditableTextareaLabelComponent),
+      multi: true,
+    },
+  ],
+})
+export class EditableTextareaLabelComponent extends EditableComponent<string> {
+  @ViewChild('textarea') textarea?: ElementRef<HTMLElement>;
+
+  textareaRows = DEFAULT_TEXTAREA_ROWS;
+
+  constructor(elementRef: ElementRef<HTMLElement>, changeDetectorRef: ChangeDetectorRef) {
+    super(elementRef, changeDetectorRef);
+  }
+
+  protected override onCancel(): void {
+    super.onCancel();
+    this.recalculateTextareaRows();
+  }
+
+  protected override onLabelClick(): void {
+    this.recalculateTextareaRows();
+    super.onLabelClick();
+    this.textarea!.nativeElement.focus();
+    this.focusedElement = this.textarea!.nativeElement;
+  }
+
+  onInput(): void {
+    this.textarea!.nativeElement.style.setProperty('height', `0`);
+    this.textarea!.nativeElement.style.setProperty(
+      'height',
+      `${this.textarea!.nativeElement.scrollHeight + 2 * BORDER_WIDTH}px`
+    );
```

Comment:
@bojidaryovchev 
1. What is the reason to set height to zero each time if it set anyway on the next step?
2. What is the reason to multiply height with BORDER_WIDTH?
3. On screen resize textarea rows are not recalculated

@neogucky 
These manipulation with height are invoked on every user input. Maybe it will be better just to make textarea resizable and srollbable, like it's done in Github comment dialog. What do you think?

Reply:
this code I found on StackOverflow, it basically handles the auto resizing of the textarea when the content grows/shrinks; so, to answer your questions:

1. it seems that by setting the height to 0 it forces the container to have a scrollbar internally even though it is never shown, so that makes possible accessing the proper value for the scrollHeight
2. we also need to include the border width in the calculations as the textarea has a border 1px therefore 1px from both top and bottom
3. the textarea rows are only calculated when we enter edit mode, actually I am pretty sure I can remove their recalculation on cancel, the idea is that by calculating them dynamically like this we achieve the effect where the content does not shift in any way when the user enters edit mode

overall it works very well the way it is, these calculations do not really affect the performance

Reply:
@neogucky

Reply:
I like how it works, the code looks a bit workarrounded but I assume that there is no simpler approach if you already checked how to best imoplement it. 

As to @dvladir comments regarding the textarea, I would leave it as it is and do a UI/UX session with customers after the release to see what improbements make sense.